hawc/apps/assessment/views.py

```diff
@@ -448,31 +449,105 @@ class AttachmentCreate(BaseCreate):
     success_message = "Attachment added."
     parent_model = models.Assessment
     parent_template_name = "parent"
+    template_name = "assessment/components/attachment_edit_row.html"
     model = models.Attachment
-    form_class = forms.AttachmentForm
+    form_class = forms.NewAttachmentForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["newAttach"] = True
+        return context
+
+    @transaction.atomic
+    def form_valid(self, form):
+        super().form_valid(form)
+        context = self.get_context_data()
+        context["object_list"] = models.Attachment.objects.get_attachments(
+            self.assessment, not context["obj_perms"]["edit"]
+        )
+        context["canEdit"] = context["obj_perms"]["edit"]
+        # context["newAttach"] = True
+        return render(self.request, "assessment/components/attachment_row.html", context)
 
 
 class AttachmentRead(BaseDetail):
     model = models.Attachment
 
     def get(self, request, *args, **kwargs):
+        if request.GET.get("test", -1) != -1:
+            return render(
+                request,
+                "assessment/components/attachment_row.html",
+                {"object": self.get_object(), "canEdit": True},
+            )
         self.object = self.get_object()
         if self.assessment.user_is_part_of_team(self.request.user):
             return HttpResponseRedirect(self.object.attachment.url)
         else:
             return PermissionDenied
 
 
+class AttachmentList(BaseList):
+    model = models.Attachment
+    parent_model = models.Assessment
+    parent_template_name = "parent"
+    template_name = "assessment/_attachment_list.html"
+    object_list = None
+    form_class = forms.NewAttachmentForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["object_list"] = models.Attachment.objects.get_attachments(
+            self.assessment, not context["obj_perms"]["edit"]
+        )
+        context["canEdit"] = context["obj_perms"]["edit"]
+        context["object"] = context["assessment"]
+        context["form"] = self.form_class()
+        return context
+
+    def get(self, request, *args, **kwargs):
+        super().get(request, *args, **kwargs)
+        context = self.get_context_data()
+        if request.GET["new"] == "True":
+            context["newAttach"] = True
+        return render(request, "assessment/_attachment_list.html", context)
+
+
 class AttachmentUpdate(BaseUpdate):
     success_message = "Assessment updated."
+    template_name = "assessment/components/attachment_edit_row.html"
     model = models.Attachment
-    form_class = forms.AttachmentForm
+    form_class = forms.NewAttachmentForm
+
+    @transaction.atomic
```

Comment:
no need for this to be atomic since it's a single write op; you need atomic is there are multiple write ops that would want to run all or none